src/NzbDrone.Core/ImportLists/ImportListSyncService.cs

```diff
@@ -0,0 +1,136 @@
+
+                reportNumber++;
+
+                var importList = _importListFactory.Get(report.ImportListId);
+
+                // Map MBid if we only have an album title
+                if (report.AlbumMusicBrainzId.IsNullOrWhiteSpace() && report.Album.IsNotNullOrWhiteSpace())
+                {
+                    var mappedAlbum = _albumSearchService.SearchForNewAlbum(report.Album, report.Artist)
+                        .FirstOrDefault();
+
+                    if (mappedAlbum == null) continue; // Break if we are looking for an album and cant find it. This will avoid us from adding the artist and possibly getting it wrong.
+
+                    report.AlbumMusicBrainzId = mappedAlbum.ForeignAlbumId;
+                    report.Album = mappedAlbum.Title;
+                    report.Artist = mappedAlbum.Artist?.Name;
+                    report.ArtistMusicBrainzId = mappedAlbum?.Artist?.ForeignArtistId;
+
+                }
+
+                // Map MBid if we only have a artist name
+                if (report.ArtistMusicBrainzId.IsNullOrWhiteSpace() && report.Artist.IsNotNullOrWhiteSpace())
+                {
+                    var mappedArtist = _artistSearchService.SearchForNewArtist(report.Artist)
+                        .FirstOrDefault();
+                    report.ArtistMusicBrainzId = mappedArtist?.ForeignArtistId;
+                    report.Artist = mappedArtist?.Name;
+                }
+
+                // Check to see if artist in DB
+                var existingArtist = _artistService.FindById(report.ArtistMusicBrainzId);
+
+                // Append Artist if not already in DB or already on add list
+                if (existingArtist == null && artistsToAdd.All(s => s.ForeignArtistId != report.ArtistMusicBrainzId))
+                {
+                    artistsToAdd.Add(new Artist
+                    {
+                        ForeignArtistId = report.ArtistMusicBrainzId,
+                        Name = report.Artist,
+                        Monitored = importList.ShouldMonitor,
+                        RootFolderPath = importList.RootFolderPath,
+                        ProfileId = importList.ProfileId,
+                        LanguageProfileId = importList.LanguageProfileId,
+                        MetadataProfileId = importList.MetadataProfileId,
+                        AlbumFolder = true,
+                        AddOptions = new AddArtistOptions{SearchForMissingAlbums = true, Monitored = importList.ShouldMonitor }
+                    });
+                }
+
+                // Add Album so we know what to monitor
+                if (report.AlbumMusicBrainzId.IsNotNullOrWhiteSpace() && artistsToAdd.Any(s=>s.ForeignArtistId == report.ArtistMusicBrainzId))
+                {
+                    artistsToAdd.Find(s => s.ForeignArtistId == report.ArtistMusicBrainzId).AddOptions.AlbumsToMonitor.Add(report.AlbumMusicBrainzId);
```

Comment:
Looks like this may be adding album to monitor even if `importList.ShouldMonitor` isn't set?